lib/plugins/json-config.js

```diff
@@ -0,0 +1,15 @@
+import File from '../file.js';
+import path from 'path';
+export default function jsonConfigPlugin (app, options) {
+  const filename = options?.file ?? 'config.json';
+  const filePath = path.isAbsolute(filename) ? new File(filename) : app.home.child(filename);
+  const fileParts = filePath.toString().split('.');
+  fileParts.splice(-1, 0, app.mode);
+  const modeFilePath = new File(fileParts.join('.'));
+  let modeConfig = {};
+  if (modeFilePath.existsSync()) {
+    modeConfig = JSON.parse(modeFilePath.readFileSync().toString());
+  }
+  const config = JSON.parse(filePath.readFileSync());
+  app.config = {...app.config, ...config, ...modeConfig};
```

Comment:
You could change the order and just use `Object.assign()` to make the code cleaner.